### ISSUE TITLE
Users unable to pass approval step in LINK asym pool deposits

### ISSUE THREAD

**Issue by purelycrickets** [shapeshift/web#6775]
### Overview

Currently in production, Users are unable to get past the approval section in an asym deposit of LINK into a pool. The user is continually prompted to approve and can continually spend gas on approvals without error messages or being able to move forward to their desired action. 

### References and additional details

Screen recording of the approval flow coninually prompting: 
https://github.com/shapeshift/web/assets/94998097/21d6a13f-7c0d-43e8-b4b9-d6e3d8032207

Previous recording was too long to upload, heres a screenshot of the multiple approvals paid without the flow continuing: 

<img width="411" alt="Screenshot 2024-04-29 at 3 31 47 PM" src="https://github.com/shapeshift/web/assets/94998097/6896363a-af8d-4c9c-8583-0e8d5a548ad4">

### Acceptance Criteria

Users are able to complete the desired flow of LINK asym deposits, or ShapeShift is able to handle errors experienced upstream in TC that could block users from coniuning their desired action.

**Comment by 0xApotheosis** [comment]
@purelycrickets did I hear you mention that you were able to replicate this on the THORSwap UI as well?

**Comment by gomesalexandre** [comment]
@purelycrickets I wasn't able to repro on THORSwap nor in the app - is this still happening?

https://github.com/shapeshift/web/assets/17035424/91ca75c3-31d5-4e32-a915-a561a88a0dca

Gut feel if this was happening both with us and THORswap is there may have been an upstream issue re: inbound addresses rotation, it would help to see at least two of those Txids to see if they approved LINK for the same spender.

**Comment by purelycrickets** [comment]
User was able to recreate in thorswap when it was first posted, but also is no longer seeing the issue now. Thanks for the explanation @gomesalexandre. 

Here are the TXIDs of the multiple approvals i made testing if it helps any: 
https://etherscan.io/tx/0x211517cfb9d490aeb1b9afba87e05e738b9e1430e96ca15916d89ef88bb35417
https://etherscan.io/tx/0xb94e029be9480368a73fe49007572841fc83c35493405ec2ba10ec6e316e59bf
https://etherscan.io/tx/0x3ef9dfdac4daf36f540d8f56235f80a01ff83e6ddfa1cb86a818b4c2e712e381
https://etherscan.io/tx/0x5f64097862aa09143a9a7f3937247833bd590c0544f85b8c6e830c7c3272c25b
https://etherscan.io/tx/0xbd3227442b97dffe54ce31c4c609703cee52fe42f234d0c63f7bbecc7a1c8e07
https://etherscan.io/tx/0x9baf428c33b0fc9f4f501d5be59e1fb65ddef6484a264f68a414e7e83075a28b
https://etherscan.io/tx/0x90ee54165a075c9f9220dcce567736346622429288a2b32b46b2096cf200fb76

**Comment by gomesalexandre** [comment]
Last one looks like it's not related to THOR and was probably inluded by mistake, but all the others look sane to me and approved the same contract as spender.

Unless this is reproducible in prod currently and after the THOR consolidation PR, I would probably disregard THORSwap having the same issue (doesn't look like anything's wrong here Tx-wise) and assume *we* had some shenanigans on allowance checks which are now fixed, close this, and possibly reopen if that ever happens again and ops can consistently repro @purelycrickets

**Comment by RussellJBennett** [comment]
This issue is persistent and repeatable. Every time. It is NOT resolved. Please acknowledge.

**Comment by RussellJBennett** [comment]
![Screenshot 2024-05-01 at 19 02 35](https://github.com/shapeshift/web/assets/168690313/7e41c722-4077-4bdf-b004-ccec4b522fff)

**Comment by RussellJBennett** [comment]
I just tried to deposit LINK into the Pool again. I'm using a Trezor One to connect and transact. 

Earlier just now, I deposited into the BTC/RUNE Pool without any issue. There is a conflict with only LINK when I try it. I've been successful before. My last LINK deposit was on 27 APR.
![Screenshot 2024-05-01 at 19 09 51](https://github.com/shapeshift/web/assets/168690313/dc486124-f069-4933-a5b5-2a01a2f24849)

**Comment by RussellJBennett** [comment]
To whom then do I escalate this issue?

I stopped using ShapeShift after several failed attempts. Because every time I tried I was charged a gas fee.

If it doesn’t work in THORswap, why would we expect a different result in Shapeshift? 

Let’s get to the root cause and resolve this.

Thx.
Russell J Bennett
9303 Buffalo Speedway
Houston, TX 77025-4417
***@***.***
(832) 588-4689 iPhone‬

On May 6, 2024, at 09:57, purelycrickets ***@***.***> wrote:


I just tried to deposit LINK into the Pool again. I'm using a Trezor One to connect and transact.

Earlier just now, I deposited into the BTC/RUNE Pool without any issue. There is a conflict with only LINK when I try it. I've been successful before. My last LINK deposit was on 27 APR.  <https://private-user-images.githubusercontent.com/168690313/327276521-dc486124-f069-4933-a5b5-2a01a2f24849.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwMDc2NDIsIm5iZiI6MTcxNTAwNzM0MiwicGF0aCI6Ii8xNjg2OTAzMTMvMzI3Mjc2NTIxLWRjNDg2MTI0LWYwNjktNDkzMy1hNWI1LTJhMDFhMmYyNDg0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwNlQxNDU1NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYzA2Mzk3ODQzY2UxMGQ2ODllNzczM2M4N2VlYzc4Y2Q0ZGQ4OTU3ZTdiNTYzYzk3NjQ2ODQwOTk4NzhkNDEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2QzMkeAkcK9t68OgkmUoXv5zkyBmAmtUdlFpTKrWst0>
Hey @RussellJBennett <https://github.com/RussellJBennett> are you trying on ShapeShift? It appears you are sharing screenshots from Thorswap, an app we have no control over the codebase of.
